zrml/global-disputes/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+    pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+    pub type OutcomeInfoOf<T> = OutcomeInfo<BalanceOf<T>, OwnerInfoOf<T>>;
+    pub type WinnerInfoOf<T> = WinnerInfo<BalanceOf<T>, OwnerInfoOf<T>>;
+    type AccountIdLookupOf<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::Source;
+    type OwnerInfoOf<T> = BoundedVec<AccountIdOf<T>, <T as Config>::MaxOwners>;
+    type LockInfoOf<T> = LockInfo<MarketIdOf<T>, BalanceOf<T>>;
+
+    #[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+    pub struct LockInfo<MarketId, Balance>(pub Vec<(MarketId, Balance)>);
+
+    impl<MarketId, Balance> Default for LockInfo<MarketId, Balance> {
+        fn default() -> Self {
+            Self(Vec::new())
+        }
+    }
+
+    impl<MarketId, Balance> MaxEncodedLen for LockInfo<MarketId, Balance>
+    where
+        Balance: MaxEncodedLen,
+        MarketId: MaxEncodedLen,
+    {
+        fn max_encoded_len() -> usize {
+            MarketId::max_encoded_len().saturating_add(Balance::max_encoded_len())
+        }
+    }
+
+    #[derive(Debug, TypeInfo, Decode, Encode, MaxEncodedLen, Clone, PartialEq, Eq)]
+    pub struct OutcomeInfo<Balance, OwnerInfo> {
+        pub outcome_sum: Balance,
+        pub owners: OwnerInfo,
+    }
+
+    #[derive(TypeInfo, Decode, Encode, MaxEncodedLen, Clone, PartialEq, Eq)]
+    pub struct WinnerInfo<Balance, OwnerInfo> {
+        pub outcome: OutcomeReport,
+        pub vote_sum: Balance,
+        pub is_finished: bool,
+        pub owners: OwnerInfo,
+    }
+
+    impl<Balance: Saturating, OwnerInfo: Default> WinnerInfo<Balance, OwnerInfo> {
+        pub fn new(outcome: OutcomeReport, vote_sum: Balance) -> Self {
+            WinnerInfo { outcome, vote_sum, is_finished: false, owners: Default::default() }
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[frame_support::transactional]
+        #[pallet::weight(T::WeightInfo::add_vote_outcome(T::MaxOwners::get()))]
+        pub fn add_vote_outcome(
+            origin: OriginFor<T>,
+            #[pallet::compact] market_id: MarketIdOf<T>,
+            outcome: OutcomeReport,
+        ) -> DispatchResultWithPostInfo {
+            let sender = ensure_signed(origin)?;
+
+            let winner_info =
+                <Winners<T>>::get(market_id).ok_or(Error::<T>::NoGlobalDisputeStarted)?;
+            ensure!(!winner_info.is_finished, Error::<T>::GlobalDisputeAlreadyFinished);
+
+            ensure!(
+                <Outcomes<T>>::get(market_id, &outcome).is_none(),
+                Error::<T>::OutcomeAlreadyExists
+            );
+
+            let voting_outcome_fee = T::VotingOutcomeFee::get();
+
+            let reward_account = Self::reward_account(&market_id);
+
+            T::Currency::transfer(
+                &sender,
+                &reward_account,
+                voting_outcome_fee,
+                ExistenceRequirement::AllowDeath,
+            )?;
```

Comment:
I see your point. I like to use `reserve` for this reason too! This implies, that we need to keep track of those, who reserved funds. In addition to that it's very likely, that the outcome owner looses the funds, because there is only one winner outcome. That's why I would personally like it as it is. But I am open-minded about this.